ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -327,23 +367,33 @@ export class AccountCreation extends React.Component<AccountCreationProps, Accou
   }
 
   updateInstitutionAffiliation(attribute: string, value) {
-    this.setState(fp.set(['profile', 'institutionalAffiliations', '0', attribute], value));
+    const {requireInstitutionalVerification} = serverConfigStore.getValue();
+    if (requireInstitutionalVerification) {
+      this.setState(fp.set(['profile', 'verifiedInstitutionalAffiliation', attribute], value));
+    } else {
+      this.setState(fp.set(['profile', 'institutionalAffiliations', '0', attribute], value));
+    }
   }
 
-  showFreeTextField(option) {
-    return option === NonAcademicAffiliation.FREETEXT || option === IndustryRole.FREETEXT ||
-        option === EducationalRole.FREETEXT;
+  // TODO remove after we switch to verified institutional affiliation
+  showInstitutionAffiliationFreeTextField(option) {
+    return option === NonAcademicAffiliation.FREETEXT ||
+      option === IndustryRole.FREETEXT ||
+      option === EducationalRole.FREETEXT;
   }
 
-  clearInstitutionAffiliation() {
-    this.setState(fp.set(['profile', 'institutionalAffiliations', '0'], {
-      nonAcademicAffiliation: null,
-      role: '',
-      institution: '',
-      other: ''
-    }));
+  // cannot destructure because verifiedInstitutionalAffiliation may not be defined
+  // todo is this a code smell?
```

Comment:
am I doing crimes?

Reply:
This will still be needed because the value is null/undefined until `renderVerifiedInstitutionalAffiliation()` creates it.

The old flow avoids this issue by creating an empty institutional affiliation.  We could also do that here, if you think that's cleaner.